shared/utils/replaceLinks.ts

```diff
@@ -1,9 +1,11 @@
 // https://stackoverflow.com/a/3809435
+// [^\u0020-\u007F] matches non-ASCII characters (e.g. umlaute, cyrillic, arabic)
+// https://stackoverflow.com/a/150078
 const protocol = /([A-Za-z]{3,9}:(?:\/\/)?)/
 const hostname = /(?:[-;:&=+$,\w]+@)?([A-Za-z0-9-]+(?:\.[A-Za-z0-9-]+)*)/
-const path = /(?:\/(?:[+~%/.\w-]*[+~%/\w-]+)?)?/
-const query = /(?:\?([-+=&;%@.\w]*[-+=&;%@\w]+)?)?/
-const hash = /(?:#([.!/\\\w]*[!/\\\w]+)?)?/
+const path = /(?:\/(?:([+~%/.\w-]|[^\u0020-\u007F])*([+~%/\w-]|[^\u0020-\u007F]+))?)?/
+const query = /(?:\?(([-+=&;%@.\w]|[^\u0020-\u007F])*([-+=&;%@\w]|[^\u0020-\u007F])+)?)?/
+const hash = /(?:#(([.!/\\\w]|[^\u0020-\u007F])*([!/\\\w]|[^\u0020-\u007F])+)?)?/
```

Comment:
Something is not right here. When using this test string:
```
https://stackoverflow.com/a/150078
https://stackoverflow.com/a/150078
https://stackoverflow.com/a/150078
```
There should be three matches with the same url. But instead it now matches across the line break. 
Compare: 
old regex -> https://regex101.com/r/bjF2h4/1
new regex -> https://regex101.com/r/DuReg4/1

Reply:
Maybe we should use sth like `a-zA-Z0-9\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF` instead of `\w` as suggested in the comment of [this stackoverflow post](https://stackoverflow.com/a/6041965)?

Reply:
Or check out xregexp as it also supports unicode symbols natively :) https://stackoverflow.com/questions/26133593/using-regex-to-match-international-unicode-alphanumeric-characters-in-javascript